### PR TITLE
Functional test AEPTestUtils update (part 3)

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR updates the final part of functional test classes: 
1. `IdentityStateFunctionalTests.java`
2. `NetworkResponseHandlerFunctionalTests.java`
3. `NoConfigFunctionalTests.java`
4. `RestartFunctionalTests.java`
5. `SampleFunctionalTests.java`

to use the AEPTestUtils library, with the following changes:
1. Map equality logic is updated to use JSON comparison APIs
2. Remove usages of Map/bytes `flatten`
6. Remove local versions of test util classes
7. Make sure Java collection `.get()` is protected with size checks
8. Update to use TestableNetworkRequest from NetworkService methods

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edge/src/androidTest/java/com/adobe/marketing/mobile/EdgeFunctionalTests.java [hunk 1]
```diff
@@ -810,31 +930,38 @@ public void testSendEvent_twoConsecutiveCalls_appendsReceivedClientSideStore() t
...
+			"        ]" +
+			"      }" +
+			"  }" +
+			"}";
+		assertExactMatch(
+			expected,
+			networkRequest.getBodyJson(),
+			new CollectionEqualCount(Subtree, "meta.state.entries"),
+			new AnyOrderMatch("meta.state.entries"),
+			new ElementCount(18, Subtree)
```
Comment (cacheung) [inline code/edge/src/androidTest/java/com/adobe/marketing/mobile/EdgeFunctionalTests.java:963]: Do we need // Asserting body has 18 total key value elements here?

File: code/edge/src/androidTest/java/com/adobe/marketing/mobile/EdgeFunctionalTests.java [hunk 2]
```diff
@@ -876,33 +1003,42 @@ public void testSendEvent_twoConsecutiveCalls_resetBefore_appendsReceivedClientS
...
```
Comment (cacheung) [inline code/edge/src/androidTest/java/com/adobe/marketing/mobile/EdgeFunctionalTests.java:1040]: Do we need // Asserting body has 18 total key value elements here?

File: code/edge/src/androidTest/java/com/adobe/marketing/mobile/NetworkResponseHandlerFunctionalTests.java [hunk 3]
```diff
@@ -391,25 +408,34 @@ public void testProcessResponseOnError_WhenTwoEventJsonError_dispatchesTwoEvents
...
+			event1.getUniqueIdentifier() +
+			"\"," +
+			"  \"requestId\": \"" +
+			requestId +
+			"\"," +
+			"  \"status\": 2003," +
+			"  \"title\": \"Failed to process personalization event\"," +
+			"  \"type\": \"personalization\"" +
+			"}";
+		JSONAsserts.assertEquals(expectedEventData2, dispatchEvents.get(1).getEventData());
```
Comment (addb) [inline code/edge/src/androidTest/java/com/adobe/marketing/mobile/NetworkResponseHandlerFunctionalTests.java:438]: assertEquals also compares agains the expected right? Does it atleast compare the size (no of keys) for both expected vs actual Also we could take scope into account?

Comment (addb) [reply code/edge/src/androidTest/java/com/adobe/marketing/mobile/NetworkResponseHandlerFunctionalTests.java:438]: I believe assertEquals should have 2 way equality by default and one simple addition would be to by default compare sizes to start with if we are not doing it currently.

**Review by cacheung — APPROVED** [review]

File: code/edge/src/androidTest/java/com/adobe/marketing/mobile/RestartFunctionalTests.java [hunk 4]
```diff
@@ -56,6 +57,11 @@ public void setup() throws Exception {
 		resetTestExpectations();
 	}
 
+	@After
+	public void tearDown() {
+		mockNetworkService.reset();
```
Comment (cacheung) [inline code/edge/src/androidTest/java/com/adobe/marketing/mobile/RestartFunctionalTests.java:62]: do we need to update this to resetTestExpectation?

**Pull request merged**